src/Symfony/Component/DependencyInjection/Argument/ServiceLocator.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Argument;
```

Comment:
Just asking, should it really be in this namespace? Seems a bit confusing to me. Can't we just keep it in `Symfony\Component\DependencyInjection` and find an appropriate name?

Reply:
like RewindableIterator: this is internal class, I don't think it should be at the root of the component

Reply:
Fair enough. But this isn't an argument in the sense of the `ArgumentInterface`, i.e a representation used before resolution/compilation. This is the actual resolved item used at runtime, indeed like the `RewindableGenerator`. As both are internal, we could move them to a Resolved/Runtime` sub-namespace, just in order to not bring confusion with actual `ArgumentInterface` implementations which are "descriptors". Those ones can be used by anyone.

Reply:
Still not convinced this deserved more. Not all classes in a subnamespace have to implement the same interface.
Here,if there is a rule, it's that all classes that have the  `Argument` suffix should implement `ArgumentInterface`.
But I wouldn't bug on some others like this one...

Reply:
No problem, just random thoughts.